doc/service-tests.md

```diff
@@ -7,255 +7,249 @@ automated tests should be included. They serve three purposes:
 
-Since we don't have an easy way to get the server to return a real repository
-request without a Content-Disposition header, we will intercept the request
-and provide our own mock response. We use the `intercept()` method provided by
-the [icedfrisby-nock plugin][icedfrisby-nock]. It takes a setup function,
-which returns an interceptor, and exposes the full API of the HTTP mocking
-library [Nock][].
+```js
+t.create('Build status (private application)')
+  .get('/build/wercker/blueprint.json')
+  .expectJSON({ name: 'build', value: 'private application not supported' })
+```
+
+## (5) Mocking Responses
+
+If we didn't have a stable example of a private project, another approach would be to mock the response. An alternative test for the 'private application' case might look like:
 
 ```js
-t.create('missing content-disposition header')
-  .get('/foo/bar.json')
-  .intercept(nock => nock('https://api.travis-ci.org')
-    .head('/foo/bar.svg')
-    .reply(200))
-  .expectJSON({ name: 'build', value: 'invalid' });
+t.create('Build status (private application)')
+  .get('/build/wercker/go-wercker-api.json')
+  .intercept(nock => nock('https://app.wercker.com/api/v3/applications/')
+    .get('/wercker/go-wercker-api/builds?limit=1')
+    .reply(401)
+  )
+  .expectJSON({ name: 'build', value: 'private application not supported' })
 ```
 
+This will intercept the request and provide our own mock response.
+We use the `intercept()` method provided by the
+[icedfrisby-nock plugin][icedfrisby-nock]. It takes a setup function,
+which returns an interceptor, and exposes the full API of the HTTP mocking
+library [Nock][].
+
 Nock is fussy. All parts of a request must match perfectly for the mock to
-take effect, including the method (in this case HEAD), scheme (https), host,
+take effect, including the HTTP method (in this case GET), scheme (https), host,
 and path.
 
 [icedfrisby-nock]: https://github.com/paulmelnikow/icedfrisby-nock#usage
 [Nock]: https://github.com/node-nock/nock
 
+Sometimes it may be also helpful to include service tests which receive a known value from the API. For example, in the `render()` method of our service, there is some logic which sets the badge color based on the build status:
+
+```js
+static render({ status, result }) {
+  if (status === 'finished') {
+    if (result === 'passed') {
+      return { message: 'passing', color: 'brightgreen' }
+    } else {
+      return { message: result, color: 'red' }
+    }
+  }
+  return { message: status }
+}
+```
+
+We can also use nock to intercept API calls to return a known response body.
+
+```js
+t.create('Build passed (mocked)')
+  .get('/build/wercker/go-wercker-api.json?style=_shields_test')
+  .intercept(nock =>
+    nock('https://app.wercker.com/api/v3/applications/')
+      .get('/wercker/go-wercker-api/builds?limit=1')
+      .reply(200, [{ status: 'finished', result: 'passed' }])
+  )
+  .expectJSON({ name: 'build', value: 'passing', colorB: '#4c1' })
```

Comment:
Service returns `brightgreen` in service code, but here we compare result with `#4c1`. What do you think about this approach? https://github.com/badges/shields/blob/7a664ca3e845c840fa299fa1ee1ee0e787593038/services/website/website.tester.js#L17

Reply:
Aah yes. I skipped over this when I first wrote it because I found the boilerplate a bit clunky and I didn't want to document it that way. Its good you raised it though. Its a bit of a tangent but I've extracted that boilerplate in a9c67ea just to make this a bit nicer to use moving forwards.